src/lib/snark_work_lib/sub_zkapp_spec.ml

```diff
@@ -0,0 +1,79 @@
+open Core_kernel
+
+[%%versioned
+module Stable = struct
+  [@@@no_toplevel_latest_type]
+
+  module V1 = struct
+    type t =
+      | Segment of
+          { statement : Transaction_snark.Statement.With_sok.Stable.V2.t
+          ; witness :
+              Transaction_snark.Zkapp_command_segment.Witness.Stable.V1.t
+          ; spec : Transaction_snark.Zkapp_command_segment.Basic.Stable.V1.t
+          }
+      | Merge of
+          { proof1 : Ledger_proof.Stable.V2.t
+          ; proof2 : Ledger_proof.Stable.V2.t
```

Comment:
So, this is the input to the function that will do the proving?

I wonder why `sok_digest` is not in this structure?
It looks like an essential input to `M.merge` and can't be derived from `proof1` and `proof2`.

Reply:
All work spec share a `sok_message` field in top level, so it's not in this abstraction level.

Here's the common fields shared for all specs.

![image](https://github.com/user-attachments/assets/003b574f-6940-4a01-8e1a-485d7a0bd927)